hail/src/main/scala/is/hail/stats/LogisticRegressionModel.scala

```diff
@@ -3,49 +3,52 @@ package is.hail.stats
 import breeze.linalg._
 import breeze.numerics._
 import is.hail.annotations.Annotation
-import is.hail.expr._
 import is.hail.expr.types._
-import is.hail.utils.FastSeq
+import is.hail.utils.{FastSeq, fatal}
 
 object LogisticRegressionTest {
-  val tests = Map("wald" -> WaldTest, "lrt" -> LikelihoodRatioTest, "score" -> ScoreTest, "firth" -> FirthTest)
+  val tests = Map("wald" -> WaldTest, "lrt" -> LikelihoodRatioTest, "score" -> LogisticScoreTest, "firth" -> LogisticFirthTest)
 }
 
-abstract class LogisticRegressionTest extends Serializable {
-  def test(X: DenseMatrix[Double], y: DenseVector[Double], nullFit: LogisticRegressionFit): LogisticRegressionTestResult[LogisticRegressionStats]
+abstract class GLMTest extends Serializable {
+  def test(X: DenseMatrix[Double], y: DenseVector[Double], nullFit: GLMFit, link: String): GLMTestResult[GLMStats]
   val schema: Type
   val emptyStats: Seq[Annotation]
 }
 
-abstract class LogisticRegressionStats {
+abstract class GLMStats {
   def toAnnotations: Seq[Annotation]
 }
 
-class LogisticRegressionTestResult[+T <: LogisticRegressionStats](val stats: Option[T], private val emptyStats: Seq[Annotation]) {
+class GLMTestResult[+T <: GLMStats](val stats: Option[T], private val emptyStats: Seq[Annotation]) {
   def toAnnotation: Annotation =
     Annotation.fromSeq(stats.map(_.toAnnotations).getOrElse(emptyStats))
 }
 
-class LogisticRegressionTestResultWithFit[T <: LogisticRegressionStats](override val stats: Option[T], private val emptyStats: Seq[Annotation], val fitStats: LogisticRegressionFit) extends LogisticRegressionTestResult[T](stats, emptyStats) {
+class GLMTestResultWithFit[T <: GLMStats](override val stats: Option[T], private val emptyStats: Seq[Annotation], val fitStats: GLMFit) extends GLMTestResult[T](stats, emptyStats) {
   override def toAnnotation: Annotation =
     Annotation.fromSeq(stats.map(_.toAnnotations).getOrElse(emptyStats) :+ fitStats.toAnnotation)
 }
 
 
-object WaldTest extends LogisticRegressionTest {
+object WaldTest extends GLMTest {
   val schema: Type = TStruct(
     ("beta", TFloat64()),
     ("standard_error", TFloat64()),
     ("z_stat", TFloat64()),
     ("p_value", TFloat64()),
-    ("fit", LogisticRegressionFit.schema))
+    ("fit", GLMFit.schema))
 
   val emptyStats: Seq[Annotation] = new Array[Annotation](schema.children.size - 1)
 
-  def test(X: DenseMatrix[Double], y: DenseVector[Double], nullFit: LogisticRegressionFit): LogisticRegressionTestResultWithFit[WaldStats] = {
+  def test(X: DenseMatrix[Double], y: DenseVector[Double], nullFit: GLMFit, link: String): GLMTestResultWithFit[WaldStats] = {
     require(nullFit.fisher.isDefined)
 
-    val model = new LogisticRegressionModel(X, y)
+    val model = link match {
+      case "logistic" => new LogisticRegressionModel(X, y)
+      case "poisson" => new PoissonRegressionModel(X, y)
+      case _ => fatal("link must be logistic or poisson")
```

Comment:
you don't need this. We will never be constructing one with the wrong arguments, since that's handled in Python